docs/appendices/release-notes/unreleased.rst

```diff
@@ -93,6 +94,13 @@ Fixes
 .. stable branch. You can add a version label (`v/X.Y`) to the pull request for
 .. an automated mergify backport.
 
+- Fixed an issue that caused the actual cast/type conversion error to be hidden
+  when it failed for a subcolumn of an object column, i.e (python).::
+
+    CREATE TABLE a (b OBJECT(DYNAMIC) AS (c REAL));
+    # create a connection and a cursor and then:
+    cursor.execute("INSERT INTO a VALUES (?)", [({"c": True},)])
+
```

Comment:
Can we simplify the example and remove python/curson execution and just keep create/insert statements?

Reply:
I rephrased it a bit, but the issue showed up only for this case, client connection and parameter replacement, if we attempt a literal `INSERT INTO a VALUE('{"c": true}')` then it goes to a special method: `ConversionException.forObjectChild(` and behaves correctly.